src/main/java/org/maxgamer/quickshop/command/subcommand/SubCommand_RemoveAll.java

```diff
@@ -56,15 +57,37 @@ public void onCommand(@NotNull CommandSender sender, @NotNull String commandLabe
                 MsgUtil.sendMessage(sender, MsgUtil.getMessage("unknown-player", null));
                 return;
             }
+
             int i = 0;
-            for (Shop shop : tempList) {
-                if (shop.getOwner().equals(shopOwner.getUniqueId())) {
-                    plugin.log("Deleting shop " + shop + " request by /qs removeall command.");
-                    shop.delete();
-                    i++;
+            if (!shopOwner.equals(sender)) { //Non-self shop
+                if (!sender.hasPermission("quickshop.removeall.other")) {
+                    MsgUtil.sendMessage(sender, MsgUtil.getMessage("no-permission", sender));
+                    return;
+                }
+                for (Shop shop : tempList) {
+                    if (shop.getOwner().equals(shopOwner.getUniqueId())) {
+                        plugin.log("Deleting shop " + shop + " request by /qs removeall command.");
+                        shop.delete();
+                        i++;
+                    }
+                }
+            } else { //Self shop
+                if (!sender.hasPermission("quickshop.removeall.self")) {
```

Comment:
Please add this permission change in plugin.yml

Reply:
added